src/base.ts

```diff
@@ -0,0 +1,271 @@
+  const documentId = containerDiv.id;
+  vw.viewerState[documentId] = {
+    pageBreaks: [],
+    canvasElements: [],
+  };
+
+  const getPageId = (pageNumber: number) => `${documentId}-page-${pageNumber}`;
+
+  const getPage = (scrollTop: number) => Math.min(
+    (vw.viewerState[documentId]?.pageBreaks.filter(b => b < scrollTop).length || 0) + 1,
+    vw.viewerState[documentId]?.pageBreaks.length || Infinity
+  );
+
+  const canvasContainer = document.createElement('div');
+  canvasContainer.className = 'canvasContainer';
+  const controlsDiv = document.createElement('div');
+  controlsDiv.className = 'viewerControls';
+  const fullPageNumberDiv = document.createElement('div');
+  fullPageNumberDiv.className = 'pageNumber';
+  const pageNumberDiv = document.createElement('div');
+  pageNumberDiv.className = 'pageNumberRaw';
+  const pageDiv = document.createElement('div');
+  const outOfDiv = document.createElement('div');
+  const pageCountDiv = document.createElement('div');
+  const nextButton = document.createElement('button');
+  nextButton.className = 'nextButton';
+  const prevButton = document.createElement('button');
+  prevButton.className = 'prevButton';
+  const zoomButton = document.createElement('button');
+  zoomButton.className = 'zoomButton';
+  const loadingTask = getDocument(documentUrl);
+
+  // handle updating page number on scroll
+  canvasContainer.onscroll = () => {
+    const pageNumber = getPage(canvasContainer.scrollTop);
+    pageNumberDiv.textContent = `${pageNumber}`;
+    if (!controlsDiv.style.opacity) {
+      controlsDiv.style.opacity = '1';
+      setTimeout(() => controlsDiv.removeAttribute('style'), 1000);
+    }
+  };
+
+  try {
+    const pdfDocument: PDFDocumentProxy = await loadingTask.promise;
+
+    // initial viewer setup
+    pageNumberDiv.textContent = '1';
+    pageDiv.textContent = 'Page ';
+    outOfDiv.textContent = '/';
+    pageCountDiv.textContent = `${pdfDocument.numPages}`;
+    fullPageNumberDiv.appendChild(pageDiv);
+    fullPageNumberDiv.appendChild(pageNumberDiv);
+    fullPageNumberDiv.appendChild(outOfDiv);
+    fullPageNumberDiv.appendChild(pageCountDiv);
+
+    nextButton.onclick = () => {
+      const pageNumber = parseInt(pageNumberDiv.textContent || '1');
+      if (pageNumber < pdfDocument.numPages) {
+        const nextPage = getPageId(pageNumber + 1);
+        canvasContainer.scrollTo({ top: (document.getElementById(nextPage)?.parentElement as HTMLElement).offsetTop });
+        pageNumberDiv.textContent = `${pageNumber + 1}`;
+      }
+    };
+    prevButton.onclick = () => {
+      const pageNumber = parseInt(pageNumberDiv.textContent || '1');
+      if (pageNumber > 1) {
+        const prevPage = getPageId(pageNumber - 1);
+        canvasContainer.scrollTo({ top: (document.getElementById(prevPage)?.parentElement as HTMLElement).offsetTop });
+        pageNumberDiv.textContent = `${pageNumber - 1}`;
+      }
+    };
```

Comment:
These are effectively the same function.
```suggestion
    const movePage = (direction: number) => () => {
      const pageNumber = parseInt(pageNumberDiv.textContent || '1');
      if (pageNumber > 1 && pageNumber < pdfDocument.numPages) {
        const nextPage = getPageId(pageNumber + direction);
        canvasContainer.scrollTo({ top: (document.getElementById(nextPage)?.parentElement as HTMLElement).offsetTop });
        pageNumberDiv.textContent = `${pageNumber + direction}`;
      }
    };

    nextButton.onclick = movePage(1);
    prevButton.onclick = movePage(-1);
```